### PR TITLE
Pictogram category

### PR DESCRIPTION
**Pull request by yohanboniface**
image

### CODE REVIEW

File: docs/administration.md [hunk 1]
```diff
@@ -10,44 +10,33 @@ Icons (aka pictograms in uMap sources) can be used in your map markers.
 
 Icons are not embedded in uMap sources, you will have to add them manually. So you can choose which icons you want to use.
 
+You can use either PNG, JPG or SVG files. SVG files are recommanded.
```
Comment (davidbgk) [inline docs/administration.md:13]:
```suggestion
You can use either PNG, JPG or SVG files. SVG files are recommended.
```

File: docs/administration.md [hunk 2]
```diff
@@ -10,44 +10,33 @@ Icons (aka pictograms in uMap sources) can be used in your map markers.
...
+
+When using SVG, it's recommanded to use icons without color. UMap will switch to white colors
```
Comment (davidbgk) [inline docs/administration.md:15]:
```suggestion
When using SVG, it's recommended to use icons without color. uMap will switch to white colors
```

File: docs/administration.md [hunk 3]
```diff
@@ -10,44 +10,33 @@ Icons (aka pictograms in uMap sources) can be used in your map markers.
...
-Go inside icons folder and remove tiny icons: `rm *-11.svg`
-
-Now, we will use imagemagick to convert svg to png.
-
-`for file in *.svg; do convert -background none $file ${file%15.svg}24.png; done`
-
-To have white icons use:
-`for file in *.svg; do convert -background none -fuzz 100% -fill white -opaque black $file ${file%15.svg}24.png; done`
-
+To import icons on your uMap server, you will need to use command `umap import_pictograms`.
```
Comment (davidbgk) [inline docs/administration.md:31]:
```suggestion
To import icons on your uMap server, you will need to use the command `umap import_pictograms`.
```

File: umap/management/commands/import_pictograms.py [hunk 4]
```diff
@@ -7,40 +7,61 @@
...
-                filepath = os.path.join(path, filename)
-                with open(filepath, 'rb') as f:
-                    picto.attribution = attribution
-                    picto.pictogram.save(filename, File(f), save=True)
-                    self.stdout.write(u"✔ Imported pictogram {filename}.".format(filename=filename))  # noqa
-
-    def extract_name(self, filename):
-        return filename[:-7].replace('-', ' ')
+                if (path.name != self.path.name):  # Subfolders only
+                    picto.category = path.name
```
Comment (davidbgk) [inline umap/management/commands/import_pictograms.py:63]:
```suggestion
                    picto.category = path.name.title()
```

Looks tempting but it doesn't work well with ALLCAPS categories (acronyms for instance).

Comment (davidbgk) [reply umap/management/commands/import_pictograms.py:63]:
```suggestion
                    if path.name.upper() != path.name:
                        picto.category = path.name.title()
```
A bit ugly and I probably miss some cases…

Comment (yohanboniface) [reply umap/management/commands/import_pictograms.py:63]: Yep, I'd better let people name the folder as they want, no ?

**Review by davidbgk — APPROVED** [review]

**Pull request merged**